driver-core/src/main/java/com/datastax/driver/core/policies/ErrorAwarePolicy.java

```diff
@@ -0,0 +1,322 @@
+         * will be considered ignored. Otherwise, the host will still be considered valid.
+         * <p/>
+         * Default value for the threshold is 1.
+         *
+         * @param errorsPerSecThreshold the inclusion threshold.
+         * @return this {@link Builder} instance, for method chaining.
+         */
+        public Builder withInclusionThreshold(double errorsPerSecThreshold) {
+            this.errorsThreshold = errorsPerSecThreshold;
+            return this;
+        }
+
+        /**
+         * Provide a class that will allow filtering errors returned by hosts.
+         * <p/>
+         * The {@link ErrorsFilter#shouldConsiderError} method will be called on every error notification the
+         * error tracker associated to the Error aware policy receives. It will then call the error filter, which
+         * will determine whether this error needs to be considered or not.
+         * <p/>
+         * The default implementation will exclude from the error counting, the following exception types :
+         * <ul>
+         *     <li>{@link ReadTimeoutException}</li>
+         *     <li>{@link WriteTimeoutException}</li>
+         *     <li>{@link UnavailableException}</li>
+         *     <li>{@link ReadFailureException}</li>
+         *     <li>{@link WriteFailureException}</li>
+         * </ul>
+         * @param errorsFilter the filter class that the policy will use.
+         * @return this {@link Builder} instance, for method chaining.
+         */
+        public Builder withErrorsFilter(ErrorsFilter errorsFilter) {
+            this.errorsFilter = errorsFilter;
+            return this;
+        }
+
+        /**
+         * Defines the time during which a host is excluded by the tracker.
+         * <p/>
+         * When a host's error rate overpasses the threshold specified in {@link #withInclusionThreshold}, the host
+         * will be considered excluded, and this for a certain period of time defined by this method. After this period,
+         * a previously excluded host will be included again unless/until it reaches the threshold again.
+         * <p/>
+         * If the error tracker, during an update, detects that a currently excluded host continues to overpass the
+         * error rate threshold, this host's exclusion time will be reset to the start.
+         * <p/>
+         * Default value for the retry period is 5 seconds.
+         *
+         * @param retryPeriod the period to use for refreshing the ErrorTracker.
+         * @return this {@link Builder} instance, for method chaining.
+         */
+        public Builder withRetryPeriod(int retryPeriod) {
+            this.retryPeriod = retryPeriod;
+            return this;
+        }
+
+        /**
+         * Creates the {@link ErrorAwarePolicy} instance.
+         * @return the newly created {@link ErrorAwarePolicy}.
+         */
+        public ErrorAwarePolicy build() {
+            return new ErrorAwarePolicy(this);
```

Comment:
Copying Andy's misplaced comment:
> Not really a problem, but I noticed with our other builders we pass the members of the builder into the constructor of the Class being built instead of the builder itself, should we do that here for consistency or does it not really matter?

It doesn't really matter, the downside is that you can't build the policy without a builder instance, but that doesn't seem to be a problem here. Actually this pattern might be a bit more convenient in cases where we allow the policy to be extended.